### ISSUE TITLE
UI Error Messages not Shown When "Model-Saving-Error" Generated in Backend

### ISSUE THREAD

**Issue by giancorderoortiz** [SAP/spartacus#9558]
(This is a bit of a grey area that requires further discussion).

**Steps to Reproduce:**
1. Navigate to SPA B2B powertools storefront.
2. Login as admin (i.e. linda wolf)
3. From My Account menu, select My Company
4. Select Budgets. 
5. Pick any budget and attempt to edit it by changing the amount to a value less than $0. 

**Expected Results :**
5. Error message on screen mentioning negative amounts are not allowed for budgets

**Actual Results:** 
5. Nothing shown. Change not made. 

----
Further comments: 

- The error returned by backend "model-saving-error" is actually generic and may appear during error conditions that are not adequately handled by the backend. 

- In order to generate meaningful error messages, additional input validation may be needed in the frontend. 

- This error was seeing many times at the beginning of the b2b occ api implementation.  Most issues seen were corrected, but this generic error still remains for some error conditions.

**Comment by giancorderoortiz** [comment]
![negative-budget](https://user-images.githubusercontent.com/46171897/97447215-b5c43e00-1905-11eb-9b56-542b9615e9e0.png)
![negative-budget-modelsavingerror](https://user-images.githubusercontent.com/46171897/97447220-b65cd480-1905-11eb-97c0-d01503c92aaa.png)

**Comment by Xymmer** [comment]
1) When customer tries to enter negative budget, we should display error. Add this validation in Spartacus so it avoids the problem. (we'll do this for 3.0)
2) If we get the error cited by GC, make sure to display an error message instead of failing silently. e.g. "Could not save". (we'll do this in future)

**Comment by Pucek9** [comment]
Probably global message is not ok. So feature depends on #9791

**Comment by Platonn** [comment]
As Bill mentioned, let's tackle in this ticket only:
> 1. When customer tries to enter negative budget, we should display error. Add this validation in Spartacus so it avoids the problem. (we'll do this for 3.0)

**Comment by Platonn** [comment]
Created separate ticket https://github.com/SAP/spartacus/issues/9801- as a follow up for
> 2. If we get the error cited by GC, make sure to display an error message instead of failing silently. e.g. "Could not save". (we'll do this in future)
